### PR TITLE
Target 3.11-rc.1 in `nightly` python build

### PR DESCRIPTION
**Pull request by scbedd**

### CODE REVIEW

**Review by mccoyp — APPROVED** [review]
Thank you for beating me to this! Should we also add a `sudo apt-get install build-essential` to the Linux pipeline, or merge this first and see if it turns out to be necessary?

**Comment by scbedd** [comment]
I see where you're going with that. I'll add it and merge this.

File: eng/pipelines/templates/jobs/tests-nightly-python.yml [hunk 1]
```diff
@@ -43,24 +48,30 @@ jobs:
...
-      vmImage: 'ubuntu-18.04'
+      name: 'azsdk-pool-mms-ubuntu-2004-general'
 
     steps:
       - template: /eng/pipelines/templates/steps/use-python-version.yml
         parameters:
           versionSpec: $(PythonVersion)
 
       - script: |
+          sudo apt-get install build-essential -y
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/tests-nightly-python.yml:64]:
```suggestion
```

File: eng/pipelines/templates/jobs/tests-nightly-python.yml [hunk 2]
```diff
@@ -22,9 +22,14 @@ jobs:
...
+          pip install setuptools==56.0.0 wheel==0.37.0 tox==3.24.3 tox-monorepo==0.1.2 packaging==21.0 requests
+          pip install $(Build.SourcesDirectory)/tools/azure-sdk-tools[build]
         displayName: Install Dependencies
 
+      - template: /eng/common/testproxy/test-proxy-tool.yml
+        parameters:
+          runProxy: false
+
       - script: |
           python ./scripts/devops_tasks/setup_execute_tests.py "$(BuildTargetingString)" --junitxml="junit/test_results.xml" --toxenv="whl"
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/tests-nightly-python.yml:34]: We will disable compatibility by passing filter type `"None"` to this parameter when it is available after another PR merges.

**Review by mccoyp — APPROVED** [review]

**Pull request merged**